### PR TITLE
Update Celeste to Julia master syntax

### PR DESCRIPTION
**Pull request by andreasnoack**
This also enables Travis testing against Julia master.

### CODE REVIEW

**Comment by andreasnoack** [comment]
Tests pass now that `DataArrays.jl` has been tagged so this is ready.

File: src/Stripe82Score.jl [hunk 1]
```diff
@@ -21,10 +21,8 @@ mag_to_flux(m)
 convert SDSS mags to SDSS flux
 """
 mag_to_flux(m::AbstractFloat) = 10.^(0.4 * (22.5 - m))
-@vectorize_1arg AbstractFloat mag_to_flux
 
 flux_to_mag(nm::AbstractFloat) = nm > 0 ? 22.5 - 2.5 * log10(nm) : NaN
-@vectorize_1arg AbstractFloat flux_to_mag
```
Comment (jeff-regier) [inline src/Stripe82Score.jl:27]: Great. I like the new dot syntax.

File: src/deterministic_vi/elbo_args.jl [hunk 2]
```diff
@@ -162,9 +162,9 @@ this helps catch them immediately.
 """
 function assert_all_finite{ParamType <: ParamSet, NumType <: Number}(
         sf::SensitiveFloat{ParamType, NumType})
-    @assert all(isfinite(sf.v)) "Value is Inf/NaNs"
-    @assert all(isfinite(sf.d)) "Gradient contains Inf/NaNs"
-    @assert all(isfinite(sf.h)) "Hessian contains Inf/NaNs"
+    @assert all(isfinite, sf.v) "Value is Inf/NaNs"
+    @assert all(isfinite, sf.d) "Gradient contains Inf/NaNs"
+    @assert all(isfinite, sf.h) "Hessian contains Inf/NaNs"
```
Comment (jeff-regier) [inline src/deterministic_vi/elbo_args.jl:167]: Nice. @gostevehoward and I were wondering about the memory use from these checks. With then new syntax, presumably temporary arrays of bools are no longer formed.

**Comment by jeff-regier** [comment]
Thanks for improvements!

**Comment by gostevehoward** [comment]
Ah, nice.

**Pull request merged**